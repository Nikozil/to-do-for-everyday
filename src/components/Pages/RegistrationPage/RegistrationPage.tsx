import React from 'react';
import { useDispatch } from 'react-redux';
import GreetingsComponent from '../../../common_components/GreetingsComponent/GreetingsComponent';
import { signUp } from '../../../Redux/modules/userSlice';
import RegistrationForm from '../../Forms/RegistrationForm/RegistrationForm';

const RegistrationPage: React.FC = () => {
  const dispatch = useDispatch();

  const handleRegistration = async (
    email: string,
    password: string,
    remember: boolean
  ) => {
    const response = await dispatch(signUp(email, password, remember));
    return response;
  };

  return (
    <GreetingsComponent>
      <RegistrationForm handleSubmit={handleRegistration} />
    </GreetingsComponent>
  );
};

export default RegistrationPage;
