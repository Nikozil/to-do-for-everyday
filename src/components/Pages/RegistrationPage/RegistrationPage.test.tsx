import { render, waitFor } from '../../../utils/tests/test-utils';
import RegistrationPage from './RegistrationPage';

describe('Login Page tests', () => {
  it('render correct', async () => {
    const { asFragment } = render(<RegistrationPage />);

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
