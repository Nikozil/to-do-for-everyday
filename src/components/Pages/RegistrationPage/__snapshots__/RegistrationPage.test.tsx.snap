// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page tests render correct 1`] = `
<DocumentFragment>
  <div
    class="greeting-page"
  >
    <div
      class="greeting-page__content"
    >
      <a
        href="/"
      >
        <div
          class="greeting-page__logo"
        >
          <img
            alt="Logo"
            src="Logo.png"
          />
        </div>
      </a>
      <form
        action="#"
      >
        <div
          class="mb-1"
        >
          <label
            class="form-label"
            for="email"
          >
            Email
          </label>
          <input
            class="form-control"
            id="email"
            name="email"
            placeholder="guest@gmail.com"
            type="email"
            value=""
          />
          <div
            class="form-text login-form__error"
          />
        </div>
        <div
          class="mb-1"
        >
          <label
            class="form-label"
            for="password"
          >
            Password
          </label>
          <input
            class="form-control"
            id="password"
            name="password"
            placeholder=""
            type="password"
            value=""
          />
          <div
            class="form-text login-form__error"
          />
        </div>
        <div
          class="mb-1 form-check"
        >
          <label
            class="form-check-label"
          >
            <input
              class="form-check-input"
              name="remember"
              type="checkbox"
              value="false"
            />
            Запомнить сессию
          </label>
        </div>
        <div
          class="d-flex flex-column mt-1 "
        >
          <button
            class="btn btn-success center-block "
            type="submit"
          >
            Зарегистрироваться
          </button>
        </div>
      </form>
    </div>
  </div>
   
</DocumentFragment>
`;
